File: FirstApp/FormView/BookMode.vb
Imports System.Data.OleDb
Public Class BookMode

    Private Sub Kittyreport_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        Dim dr_Types As OleDbDataReader = DataReader("Select DISTINCT KittyType From Kitty_Data")
        While dr_Types.Read
            If dr_Types("KittyType").GetType Is GetType(DBNull) Then Exit Sub
            Dim dr_UID As OleDbDataReader = DataReader($"Select KittyUID From Kitty_Data Where KittyType={dr_Types(0)} ORDER BY KittyNo ASC")
            Dim BookList As New List(Of Integer)

            While dr_UID.Read
                BookList.Add(dr_UID("KittyUID"))
            End While

            Dgv.Rows.Add()
            Dgv.Rows(Dgv.Rows.Count - 1).Cells(0).Value = dr_Types("KittyType").ToString.ToCurrency(AddSpace:=True)
            Dgv.Rows(Dgv.Rows.Count - 1).Cells(1).Value = $"{New Kitty(BookList.First, True).KittyNo}-{New Kitty(BookList.Last, True).KittyNo}"
            Dgv.Rows(Dgv.Rows.Count - 1).Cells(2).Value = BookList.Count
            dr_UID.Close()

            Dgv.Rows(Dgv.Rows.Count - 1).Cells(4).Value = Str(New Kitty(BookList.First).CustomerID) & "_" & Str(BookList.First)
        End While
    End Sub

    Private Sub Dgv_CellClick(sender As Object, e As DataGridViewCellEventArgs) Handles Dgv.CellClick
        If e.RowIndex >= 0 And e.ColumnIndex = 3 Then
            For Each ExistingFm As Form In Frame.MdiChildren
                If ExistingFm.Name <> "Main" Then
                    ExistingFm.Dispose()
                End If
            Next

            Dim Fm As New KittyModeCoustView With {
                .MdiParent = Frame,
                .Tag = Dgv.Rows(e.RowIndex).Cells(4).Value,
                .Dock = DockStyle.Fill
            }
            Close()
            Fm.Show()
        End If
    End Sub
End Class

File: FirstApp/FormView/BookMode.cs
using System;
using System.Collections.Generic;
using System.Data.OleDb;
using System.Linq;
using System.Windows.Forms;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

namespace FirstApp
{
    public partial class BookMode
    {
        public BookMode()
        {
            InitializeComponent();
            Dgv = _Dgv;
            _Dgv.Name = "Dgv";
        }

        private void Kittyreport_Load(object sender, EventArgs e)
        {
            OleDbDataReader dr_Types = (OleDbDataReader)ProjectModule.DataReader("Select DISTINCT KittyType From Kitty_Data");
            while (dr_Types.Read())
            {
                if (ReferenceEquals(dr_Types["KittyType"].GetType(), typeof(DBNull)))
                    return;
                OleDbDataReader dr_UID = (OleDbDataReader)ProjectModule.DataReader($"Select KittyUID From Kitty_Data Where KittyType={dr_Types[0]} ORDER BY KittyNo ASC");
                var BookList = new List<int>();

                while (dr_UID.Read())
                    BookList.Add(Conversions.ToInteger(dr_UID["KittyUID"]));

                Dgv.Rows.Add();
                Dgv.Rows[Dgv.Rows.Count - 1].Cells[0].Value = dr_Types["KittyType"].ToString().ToCurrency(AddSpace: true);
                Dgv.Rows[Dgv.Rows.Count - 1].Cells[1].Value = $"{new Kitty(BookList.First(), true).KittyNo}-{new Kitty(BookList.Last(), true).KittyNo}";
                Dgv.Rows[Dgv.Rows.Count - 1].Cells[2].Value = BookList.Count;
                dr_UID.Close();

                Dgv.Rows[Dgv.Rows.Count - 1].Cells[4].Value = Conversion.Str(new Kitty(BookList.First()).CustomerID) + "_" + Conversion.Str(BookList.First());
            }
        }

        private void Dgv_CellClick(object sender, DataGridViewCellEventArgs e)
        {
            if (e.RowIndex >= 0 & e.ColumnIndex == 3)
            {
                foreach (Form ExistingFm in My.MyProject.Forms.Frame.MdiChildren)
                {
                    if (ExistingFm.Name != "Main")
                    {
                        ExistingFm.Dispose();
                    }
                }

                var Fm = new KittyModeCoustView()
                {
                    MdiParent = My.MyProject.MyForms.Frame,
                    Tag = Dgv.Rows[e.RowIndex].Cells[4].Value,
                    Dock = DockStyle.Fill
                };
                Close();
                Fm.Show();
            }
        }
    }
}